docs: Reaction roles bot example (#3361)
discordeno/discordeno: website/docs/examples/reactionroles.md

Fantastic so now `/roles reactions create` is available, we want to add some options to the create subcommand. We will need the user to provide us with the following things:

1. Role - The role we give when the user presses the button
1. Emoji - The emoji we will put on the button.
1. Color - The color of the button.

> 1. Label - An optional label we can add to the button if the user desires.

Author: The markdown engine will increment them when rendering the text.

import roles from './roles.js'

export const commands = new Map<string, CreateApplicationCommand>(
  [roles].map(cmd => [cmd.name, cmd]),
)

export default commands
```

:::note
You can structure the command discovery in any way you might prefer, in this guide we will manually import the file and add it to the map, but you can get creative and do whatever you want. You could list all the files in a folder and after verifying that they export a command add it to this commands map, you could rely on the files to call a function that will add them to the map, you can import the map in the file and add it from there, you can do whatever you want, it is your code and so you should take your decisions
:::

> Now every time you want to add a new command to your bot you can add it in the array together with the `roles` array and it will be stored in the `commands` map.

Reviewer: Make that a separate pr if possible more clarify and less confusion is good but smaller prs are better to review

Since we never created an actual button before we will examine each property in these objects.

Let's start by the object we pass to `sendMessage`, in here we define what data the message should have, in this example we just give a really simple content and a component array, there arrays can have up to 5 objects inside and those object are `Action Rows`, we can tell that they are action rows by the `type` value they have: `MessageComponentTypes.ActionRow`, action rows also need a components array, in this array there can be up to 5 buttons, 1 select menu or 1 input text, we will se later what a select menu is and what is a input text and how to use them. Now, the buttons, we can recognize them by the `type: MessageComponentTypes.Button`, they have a different requirement from action rows, they need to have:

- A `style` - How Discord should display them,
- A `label` or `emoji` - What Discord should display as the text in the buttons

> - A `customId` - these are developer defined ids that can go up to 100 characters where we can store information and use them to tell the buttons apart from one another, we will see how this is helpful soon

Reviewer 2: ```suggestion
- A `customId` - Developer defined ID that can be up to 100 characters long. This is where we can store information and use it to tell the buttons apart from one another. We will see how this is useful soon
```

> This applies pretty much identically to the `interaction.respond` function that we call, it too has a `content` for the message and a `components` array with inside an action row that this time has 3 buttons that we define.

Reviewer 2: ```suggestion
This also applies to the `interaction.respond` function that we call. It too has a `content` for the message and a `components` array, inside of which is an action row containing 3 buttons that we defined.
```
